### DISCUSSION TITLE
HW 1 Part 3 - Question 2b and c

### DISCUSSION THREAD

**Discussion by amm20052** [cooperrc/computational-mechanics#222] [Q&A]
Part 3 
2b. I am not sure how to approximate how long a 100,000 order series would take. Any helpful insight on that would be appreciated.
2c.
Here is what i have for my code, it just won't generate a graph.
```
n = np.arange(1, 100000, 500)
N = len(n)
error = np.zeros(N, dtype=np.float32)

for i in range(N):
    approx = exptaylor(1, n[i])
    true_value = np.exp(1)
    error[i] = np.abs(approx - true_value) / true_value
    
    
plt.loglog(n, error, 'o')
plt.xlabel('Taylor Series Expansion Order')
plt.ylabel('Relative Error')
plt.title('Truncation and Roundoff Error Accumulation in log-log plot')
```

**Comment by hubert-sliwka** [comment]
For part b. I just tried a bunch of different values for my exptaylor function and from there approximated. i.e exptaylor (1,10) , exptaylor (1,100) and then noting the time differences there and approximating.

Not sure what is wrong for your part C unfortunately.

**Comment by cooperrc** [comment]
You can use the `%%timeit` command to figure out how long it take to run `N`-th order expansion, 

e.g. 

```python
%%timeit
exptaylor(1, 10)
3.59 µs ± 27.2 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

tells you that the 10-th order takes ~3.6 us. If you have two times and two orders, you can extrapolate.
